File: Modules/SMES/Tasks/SIE.Web.MES.TaskManagement/Dispatchs/Commands/CloseTaskCommand.cs
using SIE.MES.TaskManagement.Dispatchs;
using SIE.Web.Command;
using System.Linq;

namespace SIE.Web.MES.TaskManagement.Dispatchs.Commands
{
    /// <summary>
    /// 强制关闭任务单
    /// </summary>
    [JsCommand("SIE.Web.MES.TaskManagement.Dispatchs.Commands.CloseTaskCommand")]
    public class CloseTaskCommand : ViewCommand
    {
        /// <summary>
        /// 执行强制关闭操作
        /// </summary>
        /// <param name="args">args</param>
        /// <param name="scope">scope</param>
        /// <returns>强制关闭结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            string errMsg = string.Empty;
            var selectedIds = args.SelectedIds.ToList();
            errMsg = RT.Service.Resolve<DispatchController>().SetCloseTasks(selectedIds);
            if (errMsg.Length == 0)
                return "强制关闭成功";
            else
                return errMsg;
        }
    }
}

File: Modules/SMES/Tasks/SIE.Web.MES.TaskManagement/Dispatchs/Commands/CloseTaskCommand.js
SIE.defineCommand('SIE.Web.MES.TaskManagement.Dispatchs.Commands.CloseTaskCommand', {
    meta: { text: "强制关闭", group: "edit", hierarchy: "状态", iconCls: "icon-NetworkError icon-blue" },
    canExecute: function (view) {
        var selecteditems = view.getSelection();
        if (selecteditems != null && selecteditems.length > 0) {
            for (var i = 0; i < selecteditems.length; i++) {
                if (selecteditems[i].data.TaskStatus !== 40) {
                    return false;
                }
            }
            return true;
        }
        return false;
    },
    execute: function (view, source) {
        view.execute({
            withIds: true,
            selectIds: view.getSelectionIds(),
            success: function (res) { //回调
                var errMsg = res.Result;
                if (errMsg == '强制关闭成功')
                    view.reloadData();
                SIE.Msg.showMessage(errMsg);
            }
        });
    }
});